Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Node.php

```diff
@@ -69,25 +71,23 @@ public static function create(ContentSubgraphIdentity $subgraphIdentity, NodeAgg
     }
 
     /**
-     * Returns the specified property.
-     *
-     * If the node has a content object attached, the property will be fetched
-     * there if it is gettable.
+     * Returns the specified property, or null if it does not exist (or was set to null -> unset)
      *
      * @param string $propertyName Name of the property
      * @return mixed value of the property
      * @api
      */
     public function getProperty(string $propertyName): mixed
     {
-        return $this->properties[$propertyName];
+        return $this->properties->offsetGet($propertyName);
```

Comment:
is this better? after all we now access an "internal" implementation detail of the ArrayAccess implementation?

Reply:
hmm we do the same with `offsetExist`. Id argue that previously the behavior was not entirely clear - i would have assumed without the `?? null` that this array access would throw.

Besides that i think the property collection implementing `\ArrayAccess` is a bit of a hack (only for eel) and i would discourage this kind of array like access on php.

We could also discuss adding proper `has` and `get` to the `PropertyCollection` and let those get used by the node and by the `offsetGet`

Reply:
i think its an improvement. But if you prefer array access in both places or new `has` and `get` id adjust this of course ;)

Reply:
yeah, fine by me, it ust feels a bit weird :D